compiler/hash-ast/src/ast.rs

```diff
@@ -397,9 +397,39 @@ pub struct TyFnCall {
 }
 
 /// A merged type meaning that multiple types are considered to be
-/// specified in place of one, e.g. `Conv ~ Eq ~ Print`
+/// specified in place of one, e.g. `Conv ~ Eq`
 #[derive(Debug, PartialEq, Clone)]
-pub struct MergedTy(pub AstNodes<Ty>);
+pub struct MergedTy {
+    pub lhs: AstNode<Ty>,
+    pub rhs: AstNode<Ty>,
+}
+
+/// A merged type meaning that multiple types are considered to be
+/// specified in place of one, e.g. `f64 | i64`
+#[derive(Debug, PartialEq, Clone)]
+pub struct UnionTy {
+    pub lhs: AstNode<Ty>,
+    pub rhs: AstNode<Ty>,
+}
+
+/// Binary type operators enumeration.
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub enum TyOp {
```

Comment:
Maybe a better name is BinaryTyOp?

Reply:
I think to follow the same naming conventions as for expressions: `BinTyOp` is appropriate